yellowbrick/contrib/missing/bar.py

```diff
@@ -0,0 +1,247 @@
+        if self.classes_ is not None:
+            self.classes_ = np.array(classes)
+
+        # Set up classifier score visualization properties
+        if self.classes_ is not None:
+            n_colors = len(self.classes_)
+        else:
+            n_colors = None
+
+        self.colors = color_palette(kwargs.pop('colors', None), n_colors)
+
+    def get_nan_col_counts(self, **kwargs):
+        # where matrix contains strings, handle them
+        if np.issubdtype(self.X.dtype, np.string_) or np.issubdtype(self.X.dtype, np.unicode_):
+            mask = np.where( self.X == '' )
+            nan_matrix = np.zeros(self.X.shape)
+            nan_matrix[mask] = np.nan
+
+        else:
+            nan_matrix = self.X.astype(np.float)
+
+        if self.y is None:
+            nan_col_counts = [np.count_nonzero(np.isnan(col)) for col in nan_matrix.T]
+            return nan_col_counts
+
+        else:
+            # add in counting of np.nan per target y by column
+            nan_counts = []
+            for target_value in np.unique(self.y):
+
+                indices = np.argwhere(self.y == target_value)
+                target_matrix = nan_matrix[indices.flatten()]
+                nan_col_counts = np.array([np.count_nonzero(np.isnan(col)) for col in target_matrix.T])
+                nan_counts.append((target_value, nan_col_counts))
+
+            return nan_counts
+
+    def draw(self, X, y, **kwargs):
+        """Called from the fit method, this method generated a horizontal bar plot.
+
+        If y is none, then draws a simple horizontal bar chart.
+        If y is not none, then draws a stacked horizontal bar chart for each nan count per
+        target values.
+        """
+        nan_col_counts = self.get_nan_col_counts()
+
+        # the x locations for the groups
+        self.ind = np.arange(len(self.features_))
+
+        if y is None:
+            self.ax.barh(self.ind - self.width / 2, nan_col_counts, self.width,
+                            color=self.color, label=None)
+        else:
+            self.draw_stacked_bar(nan_col_counts)
+
+    def draw_stacked_bar(self, nan_col_counts):
+        """Draws a horizontal stacked bar chart with different colors
+        for each count of nan values per label.
+        """
+        for index, nan_values in enumerate(nan_col_counts):
+            label, nan_col_counts = nan_values
+
+            if index == 0:
+                # first draw should be at zero
+                bottom_chart = np.zeros(nan_col_counts.shape)
+
+            # if features passed in then, label as such
+            if self.classes_ is not None:
+                label = self.classes_[index]
+
+            color = self.colors[index]
+
+            self.ax.barh(self.ind - self.width / 2, nan_col_counts, self.width,
+                        color=color, label=label, left=bottom_chart)
+
+            # keep track of counts to build on stacked
+            bottom_chart = nan_col_counts
```

Comment:
Nice - I've always found the stacked bar charts in matplotlib a bit finicky, but I like your implementation with the loop and the bottom_chart tracking!